### PR TITLE
S3 functionality

### PR DESCRIPTION
**Pull request by gitsierra**
## Purpose

https://qmacbis.atlassian.net/browse/OY2-14767

s3 upload functionality. this works if a user is in the cognito user pool and uploads

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/components/Upload/index.tsx [hunk 1]
```diff
@@ -33,15 +34,85 @@ export const Upload = ({
     defaultValue: [],
   });
 
-  const [acceptedFiles, setAcceptedFiles] = React.useState<File[]>([]);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_acceptedFiles, setAcceptedFiles] = React.useState<File[]>([]);
```
Comment (jdinh8124) [inline services/ui-src/src/components/Upload/index.tsx:38]: I think you could remove the eslint comment and just go
  const [, setAcceptedFiles] = React.useState<File[]>([]);

File: services/ui-src/src/components/Upload/index.tsx [hunk 2]
```diff
@@ -33,15 +34,85 @@ export const Upload = ({
...
+          Promise.resolve();
+        }
+
+        return resultPromise;
+      }
       field.onChange([...field.value, ...acceptedFiles]);
+      uploadFiles(acceptedFiles).then((result: any) =>
+        field.onChange([...field.value, ...result])
+      );
+      //uploadFiles(acceptedFiles);
```
Comment (jdinh8124) [inline services/ui-src/src/components/Upload/index.tsx:115]: Think we can remove this comment

File: services/ui-src/src/components/Upload/index.tsx [hunk 3]
```diff
@@ -33,15 +34,85 @@ export const Upload = ({
...
+          const url = await Storage.vault.get(stored.key, {
+            level: "protected",
+          });
+
+          let result = {
+            s3Key: stored.key,
+            filename: fileToUpload.name,
+            contentType: fileToUpload.type,
+            url: url.split("?", 1)[0], //We only need the permalink part of the URL since the S3 bucket policy allows for public read
+            // title: fileToUpload.title,
```
Comment (jdinh8124) [inline services/ui-src/src/components/Upload/index.tsx:67]: Think this can be cleaned up

File: services/ui-src/src/components/Upload/index.tsx [hunk 4]
```diff
@@ -69,6 +140,24 @@ export const Upload = ({
...
+    const lowerCaseFileNameExtension = fileNameExtension.toLowerCase();
+
+    if (fileNameExtension === lowerCaseFileNameExtension) {
+      return file;
+    } else {
+      const updatedFileName = fileNameText.concat(lowerCaseFileNameExtension);
+      const updatedFile = new File([file], updatedFileName, {
+        type: file.type,
+      });
+      //updatedFile.name = file.title;
```
Comment (jdinh8124) [inline services/ui-src/src/components/Upload/index.tsx:156]: Last comment clean up

**Review by jdinh8124 — COMMENTED** [review]
Small change requests but it looks good! Awesome job getting the functionality working.

**Review by jdinh8124 — APPROVED** [review]

File: services/ui-src/src/components/Upload/index.tsx [hunk 5]
```diff
@@ -33,15 +34,82 @@ export const Upload = ({
     defaultValue: [],
   });
 
-  const [acceptedFiles, setAcceptedFiles] = React.useState<File[]>([]);
+  const [, setAcceptedFiles] = React.useState<File[]>([]);
 
   React.useEffect(() => {
     setAcceptedFiles([...field.value]);
   }, [field.value]);
```
Comment (benjaminpaige) [inline services/ui-src/src/components/Upload/index.tsx:41]: i dont understand why we are even doing this if we aren't using the acceptedFiles. what am i missing?

**Review by benjaminpaige — COMMENTED** [review]
looks great. just had the one question

also how do you feel about ignoring coverage for `services/ui-src/src/components/Upload/index.tsx` ; )

File: services/ui-src/src/components/Upload/index.tsx (hunk 5 above)
Comment (gitsierra) [reply services/ui-src/src/components/Upload/index.tsx:41]: this was a remnant of mapping over files that we gave to the frontend. now we map over files we get back from s3 which should match what we give it in the 1st place

**Review by jdinh8124 — APPROVED** [review]

**Pull request merged**